scipy/stats/tests/test_mstats_basic.py

```diff
@@ -1083,6 +1083,21 @@ def test_zero_division(self):
             assert_array_equal(t, np.array([np.nan, np.nan]))
             assert_array_equal(p, np.array([np.nan, np.nan]))
 
+    def test_bad_alternative(self):
+        msg = r"`alternative` must be 'less', 'greater' or 'two-sided'"
+        with pytest.raises(ValueError, match=msg):
+            mstats.ttest_ind([1, 2, 3], [4, 5, 6], alternative='foo')
+
+    @pytest.mark.parametrize("alternative", ["less", "greater"])
```

Comment:
Why is this parametrized and the tests below are not? Why is `test_bad_alternative` a separate function above but not below?

Reply:
> Why is this parametrized and the tests below are not? Why is `test_bad_alternative` a separate function above but not below?

Both are mistakes. I will parameterize and separate the two tests like I have done here.

> If we're testing `mstats`, shouldn't we test with a masked array with masked elements?

Yes, we should. I will add these in the next commit.